### PR TITLE
	Redis API: Rename options related to Redis API, describe them clearl…

### PR DESCRIPTION
**Pull request by fastio**
Rename redis_port, which the redis transport listens on for clients.

Rename redis_ssl_port, which the redis TLS transport listens on for clients.

Rename redis_database_count. Set the redis database count.

Rename redis_keyspace_replication_strategy_options. Set the replication strategy for redis keyspace.

Remove unecessary option named .

#5335 

Signed-off-by: Peng Jian <pengjian.uestc@gmail.com>

### CODE REVIEW

**Comment by fastio** [comment]
Rename `redis_transport_port` to `redis_port`, which the redis transport listens on for clients.

Rename `redis_transport_port_ssl` to `redis_ssl_port`, which the redis TLS transport listens on for clients.

Rename `redis_default_database_count` to `redis_database_count`. Set the redis database count.

Rename `redis_keyspace_options` to `redis_keyspace_replication_strategy_options`. Set the replication strategy for redis keyspace.
        
Remove unecessary option named `enable_redis_protocol`.

File: db/config.cc [hunk 1]
```diff
@@ -715,21 +715,20 @@ db::config::config(std::shared_ptr<db::extensions> exts)
...
-    , redis_keyspace_options(this, "redis_keyspace_options", value_status::Used, {}, 
-            "Enable redis protocol, list of properties of replication of redis keyspace. The available options are:\n"
-            "\n"
-            "\tclass : (Default: SimpleStrategy ).\n"
-            "\treplication_factor: (Default: 1) If the class is SimpleStrategy, set the replication factor for the strategy.\n"
-            "\tdatacenter_i: (Default: 1) If the class is NetworkTopologyStrategy, set the replication factor per datacenter.\n"
-            "\n"
-            "The properties of replication for redis keyspace.")
+    , redis_port(this, "redis_port", value_status::Used, 6379, "Port on which the REDIS transport listens for clients.")
+    , redis_ssl_port(this, "redis_ssl_port", value_status::Used, 9142, "Port on which the REDIS TLS native transport listens for clients.")
```
Comment (amoskong) [inline db/config.cc]: The FIXME comment is removed, but the problem is still there.

Do you want to share same port with native_transport_port_ssl? does it has conflict?

**Comment by nyh** [comment]
Two procedural requests:

1. The second commit has nothing to do with this pull request. I can work around this (I will commit each of them separately), but in general it's confusing.

2. Does the first commit fix issue #5335 completely and we should close it? If so, you should use the text "Fixes #5335", not just "#5335" because this causes github to automatically close this issue when the patch is committed. Moreover, you should use the text "Fixes #5335" in the individual patch's description - not in the pull request text (especially in this case when the pull request has multiple unrelated patch).

3. The second commit you added now, do it also fix an existing issue?

**Comment by fastio** [comment]
Change logs:
* DEL： Support to delete multiple keys.
* GET:    Fix the return value when the required key is not exists.
* redis-test: Add the xfailed for DEL command.

File: db/config.cc (hunk 1 above)
Comment (fastio) [reply db/config.cc]: @amoskong , Sorry. I want change the redis_ssl_port to 6389.

**Comment by fastio** [comment]
> Two procedural requests:
> 1. The second commit has nothing to do with this pull request. I can work around this (I will commit each of them separately), but in general it's confusing.

@nyh  Thanks.

> 2. Does the first commit fix issue #5335 completely and we should close it? If so, you should use the text "Fixes #5335", not just "#5335" because this causes github to automatically close this issue when the patch is committed. Moreover, you should use the text "Fixes #5335" in the individual patch's description - not in the pull request text (especially in this case when the pull request has multiple unrelated patch).

I got it. I will add the text "Fixes #5335" to the patch's description.

> 3. The second commit you added now, do it also fix an existing issue?

Yes, Fixes #5334

**Pull request closed without merging**